sway-lib-std/src/outputs.sw

```diff
@@ -268,27 +266,50 @@ pub fn output_asset_id(index: u64) -> Option<AssetId> {
 /// }
 /// ```
 pub fn output_asset_to(index: u64) -> Option<Address> {
-    match output_type(index) {
-        Some(Output::Coin) => Some(__gtf::<Address>(index, GTF_OUTPUT_COIN_TO)),
-        Some(Output::Change) => Some(__gtf::<Address>(index, GTF_OUTPUT_COIN_TO)),
-        Some(Output::Variable) => {
-            let ptr = output_pointer(index).unwrap();
+    match output_type_id(index) {
+        OUTPUT_TYPE_COIN | OUTPUT_TYPE_CHANGE => Some(__gtf::<Address>(index, GTF_OUTPUT_COIN_TO)),
+        OUTPUT_TYPE_VARIABLE => {
+            let ptr = output_pointer(index);
             Some(Address::from(ptr.add_uint_offset(OUTPUT_VARIABLE_TO_OFFSET).read::<b256>()))
         },
         _ => None,
     }
 }
 
-impl PartialEq for Output {
-    fn eq(self, other: Self) -> bool {
-        match (self, other) {
-            (Output::Coin, Output::Coin) => true,
-            (Output::Contract, Output::Contract) => true,
-            (Output::Change, Output::Change) => true,
-            (Output::Variable, Output::Variable) => true,
-            (Output::ContractCreated, Output::ContractCreated) => true,
-            _ => false,
-        }
+/// Gets the asset id and the receiver of the output at `index`.
+///
+/// # Arguments
+///
+/// * `index`: [u64] - The index of the output to get the asset id and the receiver of.
+///
+/// # Returns
+///
+/// * [Option<Address>] - The asset id and the receiver of the output.
```

Comment:
```suggestion
/// * [Option<(AssetId, Address)>] - The asset id and the receiver of the output.
```

Reply:
Good catch! I assume there will be some additional changes and optimizations, let me fix this in one of upcoming PRs.